### PR TITLE
Add materials bin tests

### PR DESCRIPTION
**Pull request by pjanik**
I was planning to use more Jasmine tests, but I finally I added mainly Cucumber integration tests.
I think in this case integration tests are even more useful, as this feature probably is not expected to be under active development (when small unit tests could help). Cucumber is slower, but I could avoid extensive mocking of everything and simulating browser events in Jasmine. I also couldn't find a pattern that I liked for Jasmine tests in this case - comparing rendered string seems to be pretty inflexible (blocks every single change in component) and has obvious limits, regexp could be too flexible and it's not very handy. I could use jQuery and write manual DOM matchers, but in that case I prefer to use Capybara.

### CODE REVIEW

**Comment by dougmartin** [comment]
Just an FYI but if you look at the react_helpers.js file I added there is a set of nice React tools for isolated testing, including simulating events.  I basically added shortcuts to these functions:

https://facebook.github.io/react/docs/test-utils.html

The pure string comparison was just there at the start to have something for Jasmine to test to make sure it worked on Travis.

**Comment by pjanik** [comment]
Right, I've noticed your helpers and even used them. But still, React helpers are pretty limited when compared to Capybara (which is not surprising, as they're designed for different scope of testing). For example I mainly wanted to test if given DOM element was rendered, visible and had some text.

Re string comparison - sure, not saying it's bad in general, it's useful in some cases. But it wouldn't work for tests I was imagining. In general I would use Jasmine for unit testing, but here I mostly wanted integration tests.

File: spec/javascripts/components/materials_bin/material_category_spec.js.coffee [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#= require helpers/react_helper
+#= require component
+
+describe 'MBMaterialsCategoryClass', ->
+  beforeEach ->
+    Portal.currentUser = {isAnonymous: true}
+
+  describe 'when loginRequired property is set to true', ->
+    it 'it should be hidden for anonymous user', ->
```
Comment (scytacki) [inline spec/javascripts/components/materials_bin/material_category_spec.js.coffee]: can you remove the duplicate `it`?

**Comment by scytacki** [comment]
This is fine to merge from my point of view.

**Comment by pjanik** [comment]
Removed unnecessary 'it', so I'm merging it.

**Pull request merged**